presto-spi/src/main/java/io/prestosql/spi/security/PrestoPrincipal.java

```diff
@@ -30,7 +29,7 @@
     public PrestoPrincipal(@JsonProperty("type") PrincipalType type, @JsonProperty("name") String name)
```

Comment:
Per the standard, an authorization identifier is just a regular identifier, so the common rules for matching identifiers apply.

```
<authorization identifier> ::=
    <role name>
  | <user identifier>

<role name> ::=
  <identifier>

<user identifier> ::=
  <identifier>
```

Given two identifiers, A and B, the rules are, basically:
* If A and B are delimited, they are equivalent if they are equal
* If A is delimited and B is not, they are equivalent if A is equal to the capitalized form of B. Viceversa if B is delimited and A is not.
* If A and B are both not delimited, they are equivalent if their capitalized forms are equal.

So, this class needs to record whether the name was originally in SQL as a delimited identifier or not.

Reply:
This is same issue like with table names. SchemaTableName should also contains such information. I am extracting this to separate issue.

Reply:
See: https://github.com/prestosql/presto/issues/313